### PR TITLE
Solutions for Chapter 1

### PR DESCRIPTION
**Pull request by visar**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -568,7 +578,10 @@ True
 >>> isVowel 'x'
 False
 -}
-isVowel c = error "isVowel: not implemented!"
+isVowel :: Char -> Bool
+isVowel c
+  | c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'y' = True
+  | otherwise = False
```
Comment (vrom911) [inline src/Chapter1.hs]: If you write it with `||`, I guess there is no much benefits of guards. You can:

```haskell
isVowel c = c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'y' 
```

We recommended guards, if you would check each case separately:

```haskell
  | c == 'a'  = True
  | c == 'e' = True
  | ...
```

File: src/Chapter1.hs [hunk 2]
```diff
@@ -632,7 +645,12 @@ Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n = x + y
+  where
+    last2 = (abs n) `mod` 100
+    x = last2 `div` 10
+    y = last2 `mod` 10
```
Comment (vrom911) [inline src/Chapter1.hs]: That is a wonderful solution! 👏🏼  You correctly noticed that it is the `div` and `mod`, cool 😎 

One hint to make your solution even shorter: you can see that you use both:

```haskell
mod (abs n) 10
div (abs n) 10
```
The standard library has the [`divMod`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:divMod) function, that actually combines inside both `div` and `mod`. And this is exactly what you use!

So you could write it this way:

```haskell
(x, y) = last2 `divMod` 10
```

You can see how we could pattern match on the pair 🙂

**Review by vrom911 — APPROVED** [review]
Awesome work on solving the challenges! ⭐ 
Congrats with finishing Chapter One 🥳 

And well done with the Advanced Task :medal_sports:

File: src/Chapter1.hs (hunk 1 above)
Comment (visar) [reply src/Chapter1.hs]: Thanks. Will do.

File: src/Chapter1.hs (hunk 2 above)
Comment (visar) [reply src/Chapter1.hs]: Awesome tip! Thanks!

**Pull request merged**